workspaces/ui/src/components/diff/v2/shape_viewers/ShapeOnlyShapeRows.js

```diff
@@ -252,7 +252,11 @@ function FieldDescription({ fieldId }) {
 
   const description = contributions.getOrUndefined(fieldId, DESCRIPTION);
 
-  const [editing, setEditing] = useState(false);
+  const [editing, _setEditing] = useState(false);
+  const setEditing = (condition) => {
+    if (process.env.REACT_APP_DONT_ALLOW_EDITING === 'true') { return; }
```

Comment:
Double negatives are confusing. I would do ...ENABLE_SPEC_EDITING = yes/no. And then the pattern we would follow in the code base is to make a context that encapsulates this concern and separates the state from the env

Reply:
my main rationale by making it "DONT ALLOW EDITING" was because I didn't want to make it something required across the entire app

Reply:
You can also achieve that by only checking if x === 'no' which is an opt in way. Either way, it should only appear once near the top when setting the react context value, and after that it would be consumed as a Boolean from react context anywhere below